Microsoft.Toolkit.Uwp.Services/Services/OneDrive/OneDriveStorageFolder.cs

```diff
@@ -426,6 +428,81 @@ public OneDriveStorageFolder(IOneDriveClient oneDriveProvider, IItemRequestBuild
+        /// <param name="successToast">The <see cref="ToastNotification"/> to be shown when download is completed</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> for the request.</param>
+        /// <returns>Task to support await of async call.</returns>
+        public async Task BackgroundDownloadAsync(StorageFolder destinationFolder, string filter = null, BackgroundTransferCompletionGroup completionGroup = null, ToastNotification successToast = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (destinationFolder == null)
+            {
+                throw new ArgumentNullException(nameof(destinationFolder));
+            }
+
+            var requestMessage = Provider.Drive.Items[OneDriveItem.Id].Request().GetHttpRequestMessage();
+            await Provider.AuthenticationProvider.AuthenticateRequestAsync(requestMessage).AsAsyncAction().AsTask(cancellationToken).ConfigureAwait(false);
+            var downloader = completionGroup == null ? new BackgroundDownloader() : new BackgroundDownloader(completionGroup);
+            downloader.TransferGroup = BackgroundTransferGroup.CreateGroup(Guid.NewGuid().ToString());
+            downloader.TransferGroup.TransferBehavior = BackgroundTransferBehavior.Serialized;
+            foreach (var item in requestMessage.Headers)
+            {
+                downloader.SetRequestHeader(item.Key, item.Value.First());
+            }
+
+            if (successToast != null)
+            {
+                downloader.SuccessToastNotification = successToast;
+            }
+
+            await Task.Run(async () => await RetriveFolderAsync(this, destinationFolder, downloader, filter, cancellationToken), cancellationToken);
+        }
+
+        /// <summary>
+        /// Method creates folder structure and starts background download for each file in given <see cref="OneDrive.OneDriveStorageFolder"/>
+        /// </summary>
+        /// <param name="onedriveFolder">OneDrive folder that will be downloaded</param>
+        /// <param name="destinationFolder">Destination <see cref="StorageFolder"/></param>
+        /// <param name="downloader"><see cref="BackgroundDownloader"/> which will be used</param>
+        /// <param name="filter">Filters the response based on a set of criteria.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> for the request.</param>
+        /// <returns>Task to support await of async call.</returns>
+        private async Task RetriveFolderAsync(OneDriveStorageFolder onedriveFolder, StorageFolder destinationFolder, BackgroundDownloader downloader, string filter = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var subfolders = await onedriveFolder.GetFoldersAsync(999, OrderBy.None, filter, cancellationToken).ConfigureAwait(false);
```

Comment:
The problem is that as for now the SDK allows to obtain 1000 items in single operations, otherwise it will throw exception. Do you have any ideas how to handle this case?